### ISSUE TITLE
circular dependencies not resolving correctly in production build

### ISSUE THREAD

**Issue by dbleier** [stealjs/steal-tools#382]
foo.js
```javascript
import Map from 'can/map/';
import 'can/map/define/'; 
import bar from 'myproject/src/bar/';

export const foo = Map.extend({
  define: {
    propA: {
      get() { return bar.attr('propA'); }
    },
    propB: {
     get() { return 'Hello World!'; }
  }
});
```
bar.js
```javascript
import Map from 'can/map/';
import 'can/map/define/'; 
import foo from 'myproject/src/foo/';

export const bar = Map.extend({
  define: {
    propA: {
      get() { return 'Hello World!';  }
    },
    propB: {
     get() { return foo.attr('propB'); }
  }
});
```
this works in development. However in production,
the define for foo calls
```javascript
var _bar = _interopRequireDefault(_myprojectBar);
```
If the define for foo is called before the define for bar, then `_bar === { 'default': {} }`

After bar is define, it seems that _bar in the foo define is not updated so that when foo.attr('propA') is called, it returns an undefined TypeError

**Comment by matthewp** [comment]
Thanks!  This is an excellent test case.

**Comment by dylanrtt** [comment]
**main.js**
```js
import foo from './foo';

// true in dev, false in production
console.log(foo.getBar().getFoo() === foo);
```

**foo.js**
```js
import bar from './bar';

export default {
    name: 'foo',
    getBar() { return bar; }
}
```

**bar.js**
```js
import foo from './foo';

export default {
    name: 'bar',
    getFoo() { return foo; }
}
```

**Comment by matthewp** [comment]
Thanks for the simplified test case.  Think we probably need https://github.com/stealjs/transpile/issues/11 for this is my guess.

**Comment by matthewp** [comment]
Fixed by #535
